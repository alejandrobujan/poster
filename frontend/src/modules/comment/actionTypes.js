export const FIND_COMMENTS_COMPLETED = "poster/comments/findComments";
export const CLEAR_COMMENTS = "poster/comments/clearComments";
export const INCREASE_PAGE = "poster/comments/increasePage";
export const DECREASE_PAGE = "poster/comments/decreasePage";
export const CLEAR_ANSWERS = "poster/comments/clearAnswers";