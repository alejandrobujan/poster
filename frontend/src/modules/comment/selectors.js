const getModuleState = state => state.comment;

export const getComments = state =>
    getModuleState(state).comments;   