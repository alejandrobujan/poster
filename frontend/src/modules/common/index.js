export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";