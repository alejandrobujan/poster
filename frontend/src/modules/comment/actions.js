import backend from '../../backend';
import * as actionTypes from './actionTypes';

const findCommentsCompleted = result => ({
    type: actionTypes.FIND_COMMENTS_COMPLETED,
    result
});

export const findComments = (postId, parentId, page) => dispatch => {
	backend.commentService.findComments(postId, page, parentId,
		elems => dispatch(findCommentsCompleted({page, elems})));
}

export const commentPost = (comment, postId, parentId, page, onErrors) => dispatch => {
	backend.commentService.commentPost(comment, postId, parentId,
		() => dispatch(findComments(postId, parentId, page)), onErrors);
}

export const clearComments = () => ({
	type: actionTypes.CLEAR_COMMENTS
});

export const previousFindCommentsResultPage = (postId, parentId, page) =>
	findComments(postId, parentId, page-1);

export const nextFindCommentsResultPage = (postId, parentId, page) =>
	findComments(postId, parentId, page+1);

export const findAnswersComment = (postId, commentId, page, onSuccess) => dispatch => {
	backend.commentService.findComments(postId, page, commentId,
		onSuccess);
}

export const commentAnswer = (comment, postId, id, page, onSuccess, onErrors) => dispatch => {
	backend.commentService.commentPost(comment, postId, id,
		() => dispatch(findAnswersComment(postId, id, page, onSuccess)), onErrors);
}

export const nextFindAnswersResultPage = (postId, parentId, page, onSuccess) =>
	findAnswersComment(postId, parentId, page+1, onSuccess);
	



