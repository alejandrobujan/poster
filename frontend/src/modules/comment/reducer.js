import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    comments: {
        page: 0,
        elems: {
            items: [],
            existMoreItems: false
        }
    }
};

const comments = (state = initialState.comments, action) => {
    switch(action.type) {

        case actionTypes.FIND_COMMENTS_COMPLETED:
            return action.result;
            
        case actionTypes.CLEAR_COMMENTS:
            return initialState.comments;   

        default:
            return state;
    }
}



const reducer = combineReducers({
    comments
});

export default reducer;