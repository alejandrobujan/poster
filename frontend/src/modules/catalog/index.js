import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


