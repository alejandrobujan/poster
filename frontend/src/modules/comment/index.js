import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};