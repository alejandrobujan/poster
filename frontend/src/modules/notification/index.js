import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


