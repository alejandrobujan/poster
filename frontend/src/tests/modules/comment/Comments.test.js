import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import store from "../../../store";

import { Comments } from "../../../modules/comment";

describe("Comments", () => {

    const user = {
        avatar: null,
        userName: "Pepe"
    }

    const emptyComments = [];
    const oneComment = [{id: 1, description: "Hola", date: 10382349823, answers: 2, parentId: null, level: 1, userSummaryDto: user}];

    it("renders correctly", () => {
		const tree = renderer
        .create(
				<Provider store={store}>
					<MemoryRouter>
						<Comments comments={emptyComments} postId={1}/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

    it("empty comments", async () => {
		render(
			<Provider store={store}>
				<MemoryRouter>
                    <Comments comments={emptyComments} postId={1}/>
				</MemoryRouter>
			</Provider>
		);

		const noComments = screen.getByTestId('NoComments');

	});

    it("not empty comments", async () => {
		render(
			<Provider store={store}>
				<MemoryRouter>
                    <Comments comments={oneComment} postId={1}/>
				</MemoryRouter>
			</Provider>
		);

		const comments = screen.getByTestId('ManyComments');

	});
    
});