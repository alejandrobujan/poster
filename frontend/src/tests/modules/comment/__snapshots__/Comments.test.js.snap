// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders correctly 1`] = `
<div
  className="alert alert-danger"
  data-testid="NoComments"
  role="alert"
>
  No comments found
</div>
`;
